odoo/fields.py

```diff
@@ -2376,6 +2376,32 @@ def __init__(self, comodel_name=Default, relation=Default, column1=Default,
             **kwargs
         )
 
+    def _get_mro_has_field_relation(self, model, name):
+        for mro in type(model).__mro__:
+            if hasattr(mro, '_abstract') and\
+                    not mro._abstract and name in mro.__dict__:
+                field = mro.__dict__[name]
+                return self.relation == field.args.get('relation')
+        return False
+
+    def _check_relation(self, model, name):
+        if isinstance(model._inherit, str):
+            inheriteds = [model._inherit]
```

Comment:
it's insufficient this check. If you have:

```python
In a module 1:
class A(models.Model):
	_name = 'a'
	field = fields.Many2many(...)

class B(models.Model):
	_name = 'b'
	_inherit = 'a'

In a module 2 that depends of module 1:
class B(models.Model):
	_inherit = 'b'
```
If you have installed module 2, then model b has _inherit = 'b' instead of 'a'

Reply:
Agree.